src/main.rs

```diff
@@ -443,9 +443,20 @@ async fn serprog_task(mut class: CdcAcmClass<'static, CustomUsbDriver>, r: SpiRe
                     continue;
                 }
                 log::debug!("Received SSpiFreq packet: {:?}", buf);
-                let freq = u32::from_le_bytes([buf[0], buf[1], buf[2], buf[3]]);
-                spi.set_frequency(freq);
-                if let Err(e) = class.write_packet(&[S_ACK]).await {
+                let try_freq = u32::from_le_bytes([buf[0], buf[1], buf[2], buf[3]]);
+                log::debug!("Setting SPI frequency: {:?}", try_freq);
+                spi.set_frequency(try_freq);
+                let actual_freq = try_freq.to_le_bytes(); // TODO can we report what the hardware has set up?
```

Comment:
I suppose the frequency is stored in the config struct which we could read back but I doubt that has any benefit over just returning the requested frequency.

Reply:
Well with divisors the actual value is a discrete one as close as possible to the requested one...  Not really important to know the precise value indeed